tst/EnergyPlus/unit/WaterManager.unit.cc

```diff
@@ -0,0 +1,129 @@
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+
+// Google Test Headers
+#include <gtest/gtest.h>
+
+// EnergyPlus Headers
+#include <EnergyPlus/WaterManager.hh>
+#include <EnergyPlus/DataWater.hh>
+#include <EnergyPlus/ScheduleManager.hh>
+
+#include "Fixtures/EnergyPlusFixture.hh"
+
+
+using namespace EnergyPlus;
+
+TEST_F(EnergyPlusFixture, WaterManager_NormalAnnualPrecipitation)
+{
+    // This unit test ensures that the WaterManager correctly calculates the Rainfall CurrentRate
+    std::string const idf_objects = delimited_string({
+        "Site:Precipitation,",
+        "ScheduleAndDesignLevel,  !- Precipitation Model Type",
+        "0.75,                    !- Design Level for Total Annual Precipitation {m/yr}",
+        "PrecipitationSchd,       !- Precipitation Rates Schedule Name",
+        "0.80771;                 !- Average Total Annual Precipitation {m/yr}",
+
+        "Schedule:Constant,",
+        "PrecipitationSchd,",
+        ",",
+        "1;",
+    });
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    WaterManager::GetWaterManagerInput();
+
+    ScheduleManager::Schedule.allocate(1);
+    ScheduleManager::Schedule(1).CurrentValue = 1.0;
+
+    WaterManager::UpdatePrecipitation();
+
+    Real64 ExpectedNomAnnualRain = 0.80771;
+    Real64 ExpectedCurrentRate = 1.0 * (0.75 / 0.80771) / DataGlobals::SecInHour;
+
+    Real64 NomAnnualRain = DataWater::RainFall.NomAnnualRain;
+    EXPECT_NEAR(NomAnnualRain, ExpectedNomAnnualRain, 0.000001);
+
+    Real64 CurrentRate = DataWater::RainFall.CurrentRate;
+    EXPECT_NEAR(CurrentRate, ExpectedCurrentRate, 0.000001);
+
+    ScheduleManager::Schedule.deallocate();
```

Comment:
This deallocate isn't necessary.  The schedule manager's state is cleared after each unit test, and the deallocate is happening over there.

Reply:
Roger that. These will be removed.